src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSender.java

```diff
@@ -66,30 +71,95 @@ public void init(@NotNull ConnectionManager connectionManager, @NotNull String e
     protected abstract int extractCountFrom(T message);
 
     @Override
-    public void send(T value) throws IOException {
+    public void send(T value) {
         Objects.requireNonNull(value, "Value for send can not be null");
 
         Counter counter = getDeliveryCounter();
         counter.inc();
         Counter contentCounter = getContentCounter();
         contentCounter.inc(extractCountFrom(value));
 
-        try {
-            ConnectionManager connection = this.connectionManager.get();
-            connection.basicPublish(exchangeName.get(), sendQueue.get(), null, valueToBytes(value), new MainMetrics(livenessMonitor, readinessMonitor));
-
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("Message sent to exchangeName='{}', routing key='{}': '{}'",
-                        exchangeName, sendQueue, toShortDebugString(value));
+        while (!canSend.get()) {
+            try {
+                Thread.sleep(1);
+            } catch (InterruptedException e) {
+                throw new IllegalStateException("Interrupted on waiting unblock send", e);
             }
```

Comment:
Please use [Condition](https://docs.oracle.com/javase/10/docs/api/java/util/concurrent/locks/Condition.html) or another monitor instead of a cycle with sleep